rmf_fleet_adapter_python/scripts/rmf_fleet_adapter_python/test_adapter.py

```diff
@@ -0,0 +1,594 @@
+            0.01,
+            partial(self._timer_cb,
+                    waypoints=waypoints,
+                    next_arrival_estimator=next_arrival_estimator,
+                    path_finished_callback=path_finished_callback)
+        )
+
+    def stop(self):
+        self.timer.reset()
+        self.timer.cancel()
+
+    def dock(self, dock_name, docking_finished_callback):
+        assert dock_name in self.dock_to_wp
+
+        # For both dockings and visited waypoints, increment the associated
+        # keyed values by 1. Or start it off at 1 if it doesn't exist yet.
+        self.dockings[dock_name] = self.dockings.get(dock_name, 0) + 1
+
+        waypoint = self.dock_to_wp[dock_name]
+        self.visited_waypoints[waypoint] = \
+            self.visited_waypoints.get(waypoint, 0) + 1
+
+        docking_finished_callback()
+        print("[RobotCommandHandle] DOCKING FINISHED")
+
+    def _timer_cb(self,
+                  waypoints,
+                  next_arrival_estimator,
+                  path_finished_callback):
+        if not self.active:
+            return
+
+        if self.current_waypoint_target < len(waypoints):
+            self.current_waypoint_target += 1
+
+        if self.updater:
+            # This waypoint is a plan waypoint, NOT graph waypoint!!
+            previous_waypoint = waypoints[self.current_waypoint_target - 1]
+
+            if previous_waypoint.graph_index.has_value:
+                print("[RobotUpdateHandle] UPDATING ROBOT POSITION:",
+                      previous_waypoint.graph_index.value)
+
+                self.updater.update_position(
+                    previous_waypoint.graph_index.value,
+                    previous_waypoint.position[2]
+                )
+                self.visited_waypoints[previous_waypoint.graph_index.value] = (
+                    self.visited_waypoints
+                        .get(previous_waypoint.graph_index.value, 0)
+                    + 1
+                )
+            else:
+                print("[RobotUpdateHandle] UPDATING ROBOT POSITION DEFAULT:",
+                      previous_waypoint.position)
+                # TODO(CH3): NOTE(CH3): Confirm this magic string is wanted
+                self.updater.update_position("test_map",
+                                             previous_waypoint.position)
+
+        if self.current_waypoint_target < len(waypoints):
+            # Again, this waypoint is a plan waypoint! NOT a graph waypoint!!
+            waypoint = waypoints[self.current_waypoint_target]
+            test_delay = (datetime.timedelta(milliseconds=750)
+                          * self.current_waypoint_target)
+
+            now = datetime.datetime.fromtimestamp(self.node
+                                                  .get_clock()
+                                                  .now()
+                                                  .nanoseconds
+                                                  / 1e9)
```

Comment:
it would be better to store this entire parameter into a variable for readability